src/main/ServiceMethod.js

```diff
@@ -1,260 +1,261 @@
+             * @param reader {circuits.ServiceDescriptorReader} - the reader used to obtain information about the service.
+             * @param provider {circuits.DataProvider} - the object providing low-level data access.
+             */
+            constructor: function (methodName, reader, provider) {
+                this.name = methodName;
+                this.reader = reader;
+                this.provider = provider;
+                this.responsePayloadName = reader.getResponsePayloadName(this.name);
+                this.requestPayloadName = reader.getRequestPayloadName(this.name);
+                this.transport = reader.getMethodTransport(this.name);
+                this.smdMethod = reader.getMethod(this.name);
+            },
+
+            /**
+             * Call this service method with the provided params and plugins.
+             * Note: that this is not generally expected to be called directly by developers - see the Service
+             * class for an example of how it is wrapped up in a service-specific named method.
+             * @param {Object} - the parameters to the service call, these are service-call specific as defined by the SMD this
+             *  service is based on. There is special handling for request payloads (e.g. for POST's and PUT's):
+             *  - If params.payload exists it is used as the initial payload, otherwise params is used.
+             *  - If the initial payload does not have the structure expected by the type defined in the SMD, it is coerced
+             *      into that structure before processing.
+             * @param {hash by type of arrays of circuits.Plugin's} - the plugins that should be applied to this invocation.
+             */
+            invoke: function (params, plugins) {
+                logger.debug("Calling service method: " + this.name + " with params", params);
+
+                var that = this,
+                    provider = this.provider,
+                    method = this.transport,
+                    url = this.reader.getServiceUrl(this.name, params),
+                    jsonpCallbackParam = this.reader.getJsonpCallbackParameter(),
+                    smdReturn = this.reader.getResponseSchema(this.name),
+                    payloadParamDef = this.reader.getRequestPayloadParam(this.name),
+                    headers,
+                    requestPayload = params.payload || params,
+                    newParams = util.mixin({}, params),
+                    requestParams,
+                    responseType = (smdReturn.type === "any" ? "blob" : "json"),
+                    timeout = this.reader.getMethodTimeout(this.name),
+                    providerHandler = function (statusCode, data, ioArgs) {
+                        var status = statusCode.toString(),
+                            payload = (statusCode < 400) ? that.processResponse(statusCode, data, plugins, ioArgs) : data,

```

Comment:
Actually, `data` looks unchanged within `processResponse` - it only extracts the payload object. So anywhere within this `providerHandler` method, you should be able to attach the data as the response property (i.e., I think it would be `newParams.response = data`).